src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -620,117 +703,205 @@ def particles_on_grid(self):
+        x_results, v_results = self._integrator.push(
+            pos_tracked, vel_tracked, B, E, self.q, self.m, self.dt
+        )
+
+        self.x[self._tracked_particle_mask], self.v[self._tracked_particle_mask] = (
+            x_results,
+            v_results,
+        )
+
+    def _update_velocity_stopping(self, summed_field_values):
+        current_speeds = np.linalg.norm(
+            self.v[self._tracked_particle_mask], axis=-1, keepdims=True
+        )
+        velocity_unit_vectors = np.multiply(
+            1 / current_speeds, self.v[self._tracked_particle_mask]
+        )
+        dx = np.multiply(current_speeds, self.dt)
+
+        stopping_power = np.zeros((self.nparticles_tracked, 1))
+        relevant_kinetic_energy = (
+            self._particle_kinetic_energy[self._tracked_particle_mask] * u.J
+        )
+
+        for cs in self._stopping_power_interpolators:
+            interpolation_result = (
+                cs(relevant_kinetic_energy).to(u.J * u.m**2 / u.kg).value
+            )
+
+            stopping_power += interpolation_result
+
+        # Take the negative stopping power since we want to be removing energy
+        # from the particles
+
+        energy_loss_per_length = np.multiply(
+            stopping_power,
+            summed_field_values["rho"].to(u.kg / u.m**3).value[:, np.newaxis],
+        )
+
+        dE = -np.multiply(energy_loss_per_length, dx)
+
+        # Update the velocities of the particles using the new energy values
+        # TODO: again, figure out how to differentiate relativistic and classical cases
+        E = self._particle_kinetic_energy[self._tracked_particle_mask] + dE
+
+        particles_to_be_stopped_mask = np.full(
+            shape=self._tracked_particle_mask.shape, fill_value=False
+        )
+        tracked_particles_to_be_stopped_mask = (
+            E < 0
+        ).flatten()  # A subset of the tracked particles!
+        # Of the tracked particles, stop the ones indicated by the subset mask
+        particles_to_be_stopped_mask[self._tracked_particle_mask] = (
+            tracked_particles_to_be_stopped_mask
         )
 
-        self.dt = dt
+        new_speeds = np.sqrt(2 * E / self.m)
+        self.v[self._tracked_particle_mask] = np.multiply(
+            new_speeds, velocity_unit_vectors
+        )
+
+        self._stop_particles(particles_to_be_stopped_mask)
+
+    def _push(self) -> None:
+        r"""
+        Advance particles using an implementation of the time-centered
+        Boris algorithm.
+        """
+
+        # Interpolate fields at particle positions
+        summed_field_values, E, B = self._interpolate_grid()
+
+        # Calculate an appropriate timestep (uniform, synchronized)
+        self.dt = self._update_time(summed_field_values)
+
+        # Update the position and velocities of the particles using timestep
+        # calculations as well as the magnitude of E and B fields
+        self._update_position(B, E)
```

Comment:
Maybe some slight OCD here, but it bothers me that you have "E,B" above, then "B,E" here! Try to keep the order the same so the signatures are easier to remember. 

IMO E,B is the correct order ;)